pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -289,10 +292,11 @@ func NoDiskConflict(pod *v1.Pod, meta algorithm.PredicateMetadata, nodeInfo *sch
 
 // MaxPDVolumeCountChecker contains information to check the max number of volumes for a predicate.
 type MaxPDVolumeCountChecker struct {
-	filter     VolumeFilter
-	maxVolumes int
-	pvInfo     PersistentVolumeInfo
-	pvcInfo    PersistentVolumeClaimInfo
+	filter          VolumeFilter
+	maxVolumes      int
+	pvInfo          PersistentVolumeInfo
+	pvcInfo         PersistentVolumeClaimInfo
+	volumePluginMgr *volume.VolumePluginMgr
```

Comment:
May be I am missing something but when at the time of node registration, these volume limits are already published in capacity and allocatable, why do you need to expose this again in scheduler?

Reply:
Currently the scheduler uses `volumePluginMgr` interface to identify which type of volume plugin the pod is using and what would be `VolumeLimitKey` for the plugin, so as scheduler can match that value with key/value present in node's Allocatable.

Reply:
We don't need volumePluginMgr anymore right?